app/services/doc_auth/lexis_nexis/responses/true_id_response.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module DocAuth
+  module LexisNexis
+    module Responses
+      class TrueIdResponse < LexisNexisResponse
+        def initialize(http_response)
+          super http_response
+        end
+
+        def successful_result?
+          transaction_status_passed? &&
+            product_status_passed? &&
+            doc_auth_result_passed?
+        end
+
+        def error_messages
+          return {} if successful_result?
+        end
+
+        def extra_attributes
+          true_id_product[:AUTHENTICATION_RESULT].reject do |k, _v|
+            PII_DETAILS.include? k
+          end
+        end
+
+        def pii_from_doc
+          true_id_product[:AUTHENTICATION_RESULT].select do |k, _v|
+            PII_DETAILS.include? k
+          end
+        end
+
+        private
+
+        def doc_auth_result
+          true_id_product.dig(:AUTHENTICATION_RESULT, :DocAuthResult)
+        end
+
+        def true_id_product
+          products[:TrueID]
+        end
+
+        def product_status
+          true_id_product.dig(:ProductStatus)
+        end
+
+        def detail_groups
+          %w[
+            AUTHENTICATION_RESULT
+            IDAUTH_FIELD_DATA
+            IDAUTH_FIELD_NATIVE_DATA
+            IMAGE_METRICS_RESULT
+            PORTRAIT_MATCH_RESULT
+          ].freeze
+        end
+
+        def transaction_status_passed?
+          transaction_status == 'passed'
```

Comment:
Sorry to clarify --- I think we should not even have the separate `transaction_status` method, we never use it? And same for `product_status` and `transaction_status`, they're never used. Unless we were planning to expose that key for logging? Then if so, let's just expose it?

Reply:
Ah, I see. I'm tempted to leave them for now - we haven't worked out the logging for this yet, and this will continue to evolve as we work on the error processing (this is just the happy path). I did get rid of a couple of the private methods that weren't used at all, and cleaned things up a bit.

Reply:
Yeah, I think there's some PRs coming along that do more sophisticated error handling that expect these.